File: javascript-kata/good-evil.js
function goodVsEvil(good, evil) {
    const battleResults = {
        good: "Battle Result: Good triumphs over Evil",
        evil: "Battle Result: Evil eradicates all trace of Good",
        tie: "Battle Result: No victor on this battle field"
    }


    const goodWorth = [1, 2, 3, 3, 4, 10], evilWorth = [1, 2, 2, 2, 3, 5, 10]

    const totalGoodWorth = good.split(' ').map(n => parseInt(n, 10)).reduce((sum, count, index) => sum + count * goodWorth[index], 0);
    const totalEvilWorth = evil.split(' ').map(n => parseInt(n, 10)).reduce((sum, count, index) => sum + count * evilWorth[index], 0);

    if (totalGoodWorth > totalEvilWorth) {
        return "Battle Result: Good triumphs over Evil";
    }
    else if (totalGoodWorth < totalEvilWorth) {
        return "Battle Result: Evil eradicates all trace of Good";
    }
    else {
        return "Battle Result: No victor on this battle field"
    }

}



console.log(goodVsEvil('0 0 0 0 0 10', '0 1 1 1 1 0 0'));

File: python-kata/good-evil.py
def goodVsEvil(good, evil):

    good_worth = [1, 2, 3, 3, 4, 10]
    evil_worth = [1, 2, 2, 2, 3, 5, 10] # Set the worth for both sides
    
    good_counts = list(map(int, good.split()))
    evil_counts = list(map(int, evil.split())) # Turn strings into lists
    
    total_good_worth = sum(count * worth for count, worth in zip(good_counts, good_worth)) 
    total_evil_worth = sum(count * worth for count, worth in zip(evil_counts,  evil_worth)) # create tuples out of the worth and counts and mutliply them, then get the sum
    
    if total_good_worth > total_evil_worth: return "Battle Result: Good triumphs over Evil"
    elif total_good_worth < total_evil_worth: return "Battle Result: Evil eradicates all trace of Good" 
    else: return "Battle Result: No victor on this battle field"

goodVsEvil('0 0 0 0 0 10', '0 1 1 1 1 0 0')